File: Linear_Search.c
#include<stdio.h>
int i;
int a[50];
int SIZE;
int data;
int found=0;
int main()
{
    printf("Enetre the size of the array: ");
    scanf("%d",& SIZE);
    printf("Enter the element of array:\n");
    for(i=0;i<SIZE;i++)
    {
        scanf("%d",&a[i]);
    }
    printf("Enter the value you want found: ");
    scanf("%d",&data);
    for(i=0;i<SIZE;i++)
    {
        if(a[i]==data)
        {
            printf("The element found at the index:%d ",i);
            found=1;
            break;

        }
    }
    if(found==0)
    {
        printf("Data not found!");
    }
}

File: Linear_Search.java
import java.util.Scanner;

public class Linear_Search {

    // Function to perform linear search
    public static int linearSearch(int[] arr, int target) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == target) {
                return i; // target found at index i
            }
        }
        return -1; // target not found
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Input: Array size
        System.out.print("Enter the size of the array: ");
        int n = scanner.nextInt();

        int[] arr = new int[n];
        System.out.println("Enter " + n + " integers:");
        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }

        // Input: Target element to search
        System.out.print("Enter the target element: ");
        int target = scanner.nextInt();

        // Perform linear search
        int result = linearSearch(arr, target);

        // Output the result
        if (result == -1) {
            System.out.println("Element not found in the array.");
        } else {
            System.out.println("Element found at index " + result);
        }
    }
}
